test/functorch/test_vmap.py

```diff
@@ -4848,6 +4848,20 @@ def test_sum_scalar(self, device):
         y = vmap(lambda x: x.sum(-1))(x)
         self.assertEqual(y, x)
 
+    def test_aten_assert_metadata(self, device):
+        t = torch.tensor([0.5, 0.5], device=device)
+        with self.assertRaisesRegex(ValueError, "must only return Tensors"):
+            vmap(torch._assert_tensor_metadata)(t, device=device)
+
+        def _vmap_func(x):
+            nonlocal device
```

Comment:
Wouldn't be cleaner to just use a functools partial? Do we want the function to be able to edit the assignment of device? Or is the nonlocal device part of the test?